tests/ui/mismatched_types/mismatched-types-issue-126222.stderr

```diff
@@ -0,0 +1,51 @@
+error[E0308]: mismatched types
+  --> $DIR/mismatched-types-issue-126222.rs:7:18
+   |
+LL |             x => dbg!(x),
+   |                  ^^^^^^^ expected `()`, found integer
+   |
+   = note: this error originates in the macro `dbg` (in Nightly builds, run with -Z macro-backtrace for more info)
+help: you might have meant to return this value
+   |
+LL |             x => return dbg!(x),
+   |                  ++++++
+
+error[E0308]: mismatched types
+  --> $DIR/mismatched-types-issue-126222.rs:15:17
+   |
+LL |                 dbg!(x)
+   |                 ^^^^^^^ expected `()`, found integer
+   |
+   = note: this error originates in the macro `dbg` (in Nightly builds, run with -Z macro-backtrace for more info)
+help: you might have meant to return this value
+   |
+LL |                 return dbg!(x)
+   |                 ++++++
```

Comment:
I'm somewhat surprised: shouldn't these suggestions include a trailing `;`?

Reply:
I'm assuming this is because `dbg` expands to a block expression, right?

Reply:
> I'm somewhat surprised: shouldn't these suggestions include a trailing `;`?

Thank you.
Here I think because in the return expression after match arm the "; " is unnecessary, and the diagnostic suggestion may make error expr like "return something; ,", so in this case, I recommend using the `return something` without semicolon `;`.

Reply:
It is *idiomatic* for statements like `return` to be ended with a `;`, regardless of whether it is *valid code* or not. That being said, it doesn't matter as much, as long as the suggestions produce valid code.